src/file_uploader/mod.rs

```diff
@@ -0,0 +1,146 @@
+pub mod pinata;
+pub mod s3;
+
+use crate::file_uploader::{pinata::PinataClient, s3::S3Client};
+use crate::handlers::upload::UploadError;
+use std::path::Path;
+use std::{fs::File, io::Read};
+
+pub struct FileUploader<'a, T: PinataClient, E: S3Client> {
+    pinata_client: &'a T,
+    s3_client: &'a E,
+}
+
+impl<'a, T: PinataClient, E: S3Client> FileUploader<'a, T, E> {
+    pub fn new(pinata_client: &'a T, s3_client: &'a E) -> Self {
+        Self {
+            pinata_client,
+            s3_client,
+        }
+    }
+
+    pub async fn upload_file(&self, path: &Path) -> Result<String, UploadError> {
+        let ipfs_hash = self.pinata_client.upload_file_to_ipfs(path).await?;
+
+        // Read file contents
+        let mut file = File::open(path).map_err(|_| UploadError::OpenFile)?;
+        let mut buffer = Vec::new();
+        file.read_to_end(&mut buffer)
+            .map_err(|_| UploadError::ReadFile)?;
+
+        // Upload to S3
+        self.s3_client
+            .upload_file_to_s3(path, ipfs_hash.clone())
+            .await?;
+
+        Ok(ipfs_hash)
+    }
+}
+
+#[cfg(test)]
+pub fn get_mock_file_uploader() -> FileUploader<'static, pinata::MockPinataClient, s3::MockS3Client>
+{
+    FileUploader::new(&pinata::MockPinataClient, &s3::MockS3Client)
+}
+
+#[cfg(test)]
```

Comment:
nit: maybe put this `get_mock_file_uploader` within the `#[cfg(test)]` below (i.e. have a single `#[cfg(test)]`)

Reply:
done in https://github.com/FuelLabs/forc.pub/pull/38/commits/9472299f263e9c2b32ca2d025e07e28fe1b89510